client/src/main/java/io/pravega/client/stream/impl/Orderer.java

```diff
@@ -32,13 +33,13 @@ SegmentInputStream nextSegment(List<SegmentInputStream> segments) {
             return null;
         }
         for (int i = 0; i < segments.size(); i++) {
-            SegmentInputStream inputStream = segments.get(counter.incrementAndGet() % segments.size());
+            SegmentInputStream inputStream = segments.get(MathHelpers.abs(counter.incrementAndGet()) % segments.size());
```

Comment:
under what conditions would this be negative? When it overflows?

Just checking because this would mean we have about 2 billion segments attached to a stream. That's unless you repeatedly call this method on an arbitrary list of segments, some of which are repeated.